src/plugins/kernel_install/utils.sh

```diff
@@ -676,7 +676,7 @@ function install_kernel()
   # See shellcheck warning SC2024: sudo doesn't affect redirects. That
   # is why we use tee. Also note that the stdin is passed to the eval
   # inside cmd_manager.
-  cmd="grep -Fxq $name $INSTALLED_KERNELS_PATH"
+  cmd="${sudo_cmd}grep -Fxq $name $INSTALLED_KERNELS_PATH"
```

Comment:
Actually, the other parts of this code need to be corrected since we should not add spaces in this string concatenation. Check `sudo_cmd='sudo -E '$` at the beginning of this function.

Reply:
Oh, I see it now. It also makes sense as in case `sudo_cmd` is empty, there won't be a leading whitespace in the command.